tests/unit/core/agents/agent_components/test_streaming_debug.py

```
@@ -0,0 +1,476 @@
+        assert preview is not None
+
+    def test_part_with_non_string_content(self):
+        event = SimpleNamespace(part=SimpleNamespace(part_kind="tool-call", content=42))
+        part_type, part_kind, preview, length = _get_part_debug_data(event)
+        assert part_type == "SimpleNamespace"
+        assert part_kind == "tool-call"
+        assert preview is None
+        assert length is None
+
+
+def _make_state_manager():
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `_make_state_manager`

(ANN202)